### PR TITLE
YAML preferences

### PR DESCRIPTION
**Pull request by angelozerr**
YAML preferences

image

Signed-off-by: azerr <azerr@redhat.com>

This PR provides all settings which manages YAML LS. I set to draft because I need to test if it works correctly, but you can try it.

### CODE REVIEW

**Comment by vrubezhny** [comment]
@angelozerr Also, please, take a look at YAML JUnit tests - they fail probably because of changes in editor configuration:

```
Failures: 
  TestYaml.testSchemaExtensionPoint:91->testErrorFile:85 expected: <true> but was: <false>
  TestYaml.testSchemaExtensionPointUsingPlatformURL:96->testErrorFile:85 expected: <true> but was: <false>
```

**Review by vrubezhny — CHANGES_REQUESTED** [review]

File: org.eclipse.wildwebdeveloper/plugin.properties [hunk 1]
```diff
@@ -36,4 +36,11 @@ HTMLPreferencePage.name=HTML (Wild Web Developer)
 HTMLCompletionPreferencePage.name=Completion
 HTMLFormatPreferencePage.name=Format
 HTMLHoverPreferencePage.name=Hover
-HTMLValidationPreferencePage.name=Validation
\ No newline at end of file
+HTMLValidationPreferencePage.name=Validation
+
+# YAML
+YAMLPreferencePage.name=YAML
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.wildwebdeveloper/plugin.properties]: fixed

File: org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/ui/messages.properties [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# *
+# * Contributors:
+# *  Angelo ZERR (Red Hat Inc.) - initial implementation
+# *******************************************************************************/
+
+# Preference main page
+YAMLPreferencePage_SchemaAssociationsLink=See <A>'Schema association'</A> for Schema associations preferences
+YAMLPreferencePage_schemaStore_enable=Automatically pull available YAML schemas from JSON Schema Store
+YAMLPreferencePage_schemaStore_url=URL of schema store catalog to use
+YAMLPreferencePage_maxItemsComputed=The maximum number of outline symbolsand folding\nregions computed (limited for performance reasons).
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/ui/messages.properties]: fixed

File: org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/ui/messages.properties [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+# Completion page
+YAMLCompletionPreferencePage_completion=Enable/disable completion feature
+YAMLCompletionPreferencePage_suggest_parentSkeletonSelectedFirst=If true, the user must select some parent skeleton first before autocompletion starts to suggest the rest of the properties.\nWhen yaml object is not empty, autocompletion ignores this setting and returns all properties and skeletons
+YAMLCompletionPreferencePage_disableDefaultProperties=Globally set additionalProperties to false for all objects.\nSo if its true, no extra properties are allowed inside yaml.
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/ui/messages.properties]: fixed

File: org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/ui/messages.properties [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+YAMLFormatPreferencePage_format_printWidth=Specify the line length that the printer will wrap on
+
+# Hover page
+YAMLHoverPreferencePage_hover=Enable/disable hover feature
+
+# Validation page
+YAMLValidationPreferencePage_validate=Enable/disable validation feature
+YAMLValidationPreferencePage_yamlVersion=Default YAML spec version
+YAMLValidationPreferencePage_customTags=Custom tags for the parser to use
+YAMLValidationPreferencePage_disableAdditionalProperties=Globally set additionalProperties to false for all objects. So if its true, no extra properties are allowed inside yaml.
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/ui/messages.properties]: Good catch, fixed.

**Review by vrubezhny — APPROVED** [review]

**Comment by angelozerr** [comment]
Please note the YAML configuration must be improve to collect http and others ections.See https://github.com/eclipse/wildwebdeveloper/pull/930/files#diff-35f0eb9d34c1bd751f939c11d145d2dc8d81295432566dc74b5f882f2beb5d0cR42

**Pull request merged**